### PR TITLE
Add React and React-router

### PR DESCRIPTION
**Pull request by VernL**
Create app with Create-React-App.
Create components for Home, Articles and Blog.
Add routing with React-router.
Closes issues #51 & #52.

### CODE REVIEW

**Comment by RichardLitt** [comment]
@ZJaf Want to review this PR?

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,19 @@
-node_modules/
-.DS_Store
\ No newline at end of file
+# dependencies
+/node_modules
```
Comment (RichardLitt) [inline .gitignore]: No need for the `/` in front.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,19 @@
...
+
+# testing
+/coverage
+
+# production
+/build
+
+# misc
+.DS_Store
+.env.local
```
Comment (RichardLitt) [inline .gitignore]: Let's not get into the habit of committing code which is only relevant for our own workflow. Add all of the #misc stuff to your own global .gitignore, and remove them here. Make sense?

**Comment by VernL** [comment]
> Can you integrate the map.html changes I merged recently, too?

I don't see an obvious way to insert the map <script> tags into a react component. Found a guide [here](https://www.fullstackreact.com/articles/how-to-write-a-google-maps-react-component/). Will look into it.

**Comment by ZJaf** [comment]
Hey guys, I do not see the need of using a boilerplate like create-react-app. I would rather to go through building our requirement ourselves. Especially if we are using another boilerplate like node-express! Too much ready code! for not knowing what they are about!

@Jojokats  and I we sat together today and broke down the Homepage to components and we started to put some codes together. I will setup Babel and Webpack for the project. Reading right now.

**Comment by ZJaf** [comment]
>  I don't see an obvious way to insert the map <script> tags into a react component. Found a guide [here](https://www.fullstackreact.com/articles/how-to-write-a-google-maps-react-component/). Will look into it. 

I thought we are going to use leaflet for the map ?!!! 😕

**Comment by RichardLitt** [comment]
> I thought we are going to use leaflet for the map ?!!!

Google Maps turned out to have better tutorials, so we switched. 

> Hey guys, I do not see the need of using a boilerplate like create-react-app. I would rather to go through building our requirement ourselves. Especially if we are using another boilerplate like node-express! Too much ready code! for not knowing what they are about!

Can you be more specific with the actual changes you don't like in this PR? Some boilerplates are useful, because they offload the effort of making everything ourselves. Discarding them because they are boilerplates isn't as useful as saying why this particular boilerplate doesn't suit out needs.

**Comment by VernL** [comment]
YOLO

**Comment by ZJaf** [comment]
>Can you be more specific with the actual changes you don't like in this PR? Some boilerplates are useful, because they offload the effort of making everything ourselves. Discarding them because they are boilerplates isn't as useful as saying why this particular boilerplate doesn't suit out needs. 

Just answering your question:
Sure,
 'registerserviceworker.js', why do we need this file?. why do we need a logo of react. why do we have this style.css
Why do we have a public folder? why do we use app.js and also index.js inside the src folder.  What are all those extra lines of code in .gitignore? 

However, my point was not that boilerplates are not suiting our purpose! sure they do. My assumption was that we are learning through this project and one part of learning is to set up our own react project. Both will give us a base to run our codes on it. so! yes, I am fine with both of them.

**Comment by RichardLitt** [comment]
Good questions!

There's a lot in there. 
- Take a look at the comments for registerserviceworker. It should explain why it's needed.
- Not seeing a logo of react. Do you mean the favicon? We should add and replace the one there once we have a logo. 
- The style.css is showing how to import css into .jsx. We can remove it and replace it with our already existing CSS. Want to do that? 
- Public is often what is published publicly on the server. All of the other files won't be reachable via the internet. 
- The app.js and index.js files do different the things; the first contains app logic, and the second serves as the entrypoint, and does the high-level instantiation of the app. Separating them makes it easier to separate concerns and show what is doing what. 
- I don't see any extra lines in the .gitignore - except for the newline at the end of the file, which doesn't change anything and which some editors actually demand. 

Does this answer your questions? If we don't like any of the choices made, we can change the boilerplate.

**Pull request merged**